db/dcrsqlite/sqlite.go

```diff
@@ -256,12 +266,41 @@ func NewDB(db *sql.DB) (*DB, error) {
 	d.getBlockSummaryTableInfo = fmt.Sprintf(`PRAGMA table_info(%s);`, TableNameSummaries)
 	d.getStakeInfoExtendedTableInfo = fmt.Sprintf(`PRAGMA table_info(%s);`, TableNameStakeInfo)
 
+	// Attempt to retrieve the best mainchain block heights from both tables,
+	// but fallback to the highest block if the tables have yet to be upgraded
+	// with the is_mainchain column. Log the latter case.
 	var err error
+	noSuchColPrefix := "no such column: "
 	if d.dbSummaryHeight, err = d.getBlockSummaryHeight(); err != nil {
-		return nil, err
+		// If the table scheme lacks the is_mainchain column, and upgrade is
+		// required. Fall back to querying without the is_mainchain condition.
+		ind := strings.LastIndex(err.Error(), noSuchColPrefix+"is_")
```

Comment:
This is somewhat inelegant compared to a `PRAGMA table_info` query, but it's simple and requires no extra query, just inspection of the error.